generic/sysbench.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Hariharan T.S.  <harihare@in.ibm.com>
+
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Sysbench(Test):
+
+    """
+    sysbench supports following performance tests
+    :cpu
+    :threads
+    :oltp
+    :fileio
+
+    :avocado: tags=cpu,threads
+    """
+
+    def clear_dmesg(self):
+        process.run("dmesg -c ", sudo=True)
+
+    def verify_dmesg(self):
+        self.whiteboard = process.system_output("dmesg")
+        pattern = ['WARNING: CPU:', 'Oops',
+                   'Segfault', 'soft lockup', 'Unable to handle']
+        for x in pattern:
+            if x in self.whiteboard:
+                self.fail("Test Failed : %s in dmesg" % x)
+
+    def run_cmd(self, cmdline):
+        try:
+            process.run(cmdline, ignore_status=False, sudo=True)
+        except process.CmdError as details:
+            self.fail("The sysbench failed: %s" % details)
+
+    def setUp(self):
+        softmanager = SoftwareManager()
+        if not softmanager.check_installed('sysbench') and not softmanager.install('sysbench'):
+            self.cancel("sysbench package is required to continue the test")
+        self.max_time = self.params.get('max-time', default=None)
```

Comment:
can `self.max_time` , `self.max_request` and `self.threads_locks`  has some default value as building command  with `None` not make sense as yaml file also does not have these value  , just a suggestion

Reply:
These are optional parameters, even 0 makes a sense. Hence the parameter none, if required it will be used by the tester.

Reply:
sure make sense , thanks for helping